tests/unit/versioning/VersioningRequestProcessor.spec.js

```diff
@@ -357,4 +358,67 @@ describe('test VSP', () => {
         }],
         done);
     });
+
+    it('should be able to update a null suspended version', done => {
+        let nullVersionId;
+
+        async.waterfall([next => {
+            // simulate the creation of a null suspended version.
+            const request = {
+                db: 'foo',
+                key: 'bar',
+                value: '{"qux":"quz","isNull":true}',
+                options: {
+                    versionId: '',
+                },
+            };
+            vsp.put(request, logger, next);
+        },
+        (res, next) => {
+            nullVersionId = JSON.parse(res).versionId;
+            // simulate update null version with BackbeatClient.putMetadata
+            const request = {
+                db: 'foo',
+                key: 'bar',
+                value: '{"qux":"quz2","isNull":true}',
+                options: {
+                    versioning: true,
+                    versionId: nullVersionId,
+                },
+            };
+            vsp.put(request, logger, next);
+        },
+        (res, next) => {
+            wgm.list({}, logger, next);
+        },
+        (res, next) => {
+            const expectedListing = [
+                // NOTE: should not set nullVersionId to the master version if updating a null version.
+                {
+                    key: 'bar',
+                    value: '{"qux":"quz2","isNull":true}',
+                },
+                {
+                    key: `bar\x00${nullVersionId}`,
+                    value: `{"qux":"quz","isNull":true,"versionId":"${nullVersionId}"}`,
+                },
```

Comment:
I don't understand why we have a second version here.

For me what should happen is:
- we create a null version on a suspended-versioning bucket -> creates a master key with `isNull` and an internal version ID
- we update that same null version -> master is updated with the new attributes and keeps same version ID

I don't understand why we have a version key in this case and why we end up with two different keys.

Reply:
Because of: https://scality.atlassian.net/browse/MD-843
It's a minor update, but I've upated the test to reflect the call made by putMetadata, including the addition of the versionId in the updated value.

Reply:
About the fact that we end up with a version key with an outdated value.

I am not quite sure if it is by design or not but it does not seem to be a problem. 
Once PUT object on top of this latest null version key:
- the master gets updated with the new version value,
- the null version gets updated with the old master value.

So the null version has the correct value.

Reply:
If it is a bug, I can create a following ticket for it.

Reply:
Ok good for me, indeed the duplicate key was not fixed on the 7.10 branch. We should make sure that we don't have a regression on 7.70 (I think the existing tests should cover this).